src/server/system_services/pool_controllers.js

```diff
@@ -68,88 +67,48 @@ class ManagedStatefulSetPoolController extends PoolController {
         dbg.log2(`ManagedStatefulSetPoolController::create: Pool ${this.pool_name} k8s configuration is`,
             JSON.stringify(pool_k8s_conf, null, 2));
 
-        await kube_utils.apply_conf(pool_k8s_conf);
-        return yaml_utils.stringify(pool_k8s_conf);
+        await KubeStore.instance.create_secret(secret_k8s_conf);
+        await KubeStore.instance.create_backingstore(pool_k8s_conf);
```

Comment:
Why not paralleling these operations? `Promise.all`

Also what happens if the secret failed but the backingsotre succedes. I think you should retry or at least bailout by deleting the backing store.

Reply:
because we need the secret first. the order is important. 
if it will fail we will throw exception so the backingstore won't be created.